### PR TITLE
refactor: flake8 fixes for conftest files

### PR DESCRIPTION
**Pull request by synchon**
* [ ] fix #(issue number)
* [x] description of feature/fix
* [x] tests added/passed
* [ ] add an entry to the [next release](../release-notes/next-release.md)

This PR resolves `flake8` issues for `conftest.py` files required for `pytest`. Also, has type annotations and code style upgrades wherever required.

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -1,103 +1,125 @@
...
+from cobra import Metabolite, Model, Reaction, Solution
 from cobra.io import read_sbml_model
 from cobra.util import solver as sutil
 
 
 try:
     import pytest
-    import pytest_benchmark
 except ImportError:
     pytest = None
```
Comment (Midnighter) [inline tests/conftest.py]: I think you can just convert this whole thing to a simple `import pytest`. We run our test suite with pytest, we have to assume it's installed.

**Review by Midnighter — APPROVED** [review]
So how will you celebrate when all code passes flake8? 🙂

File: tests/conftest.py (hunk 1 above)
Comment (synchon) [reply tests/conftest.py]: I agree.

**Comment by synchon** [comment]
A good beer probably lol! 😂

**Pull request merged**